src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala

```diff
@@ -112,23 +117,175 @@ object DockerBuildPlugin extends AutoPlugin {
+    if (dockerImageNamePrefix.value.nonEmpty) {
+      dockerImageRegistryHost.value + '/' + dockerImageNamePrefix.value + '/' +
+        dockerImageName.value
+    } else {
+      dockerImageRegistryHost.value + '/' + dockerImageName.value
+    }
+  }
+
+  /** The full name of the dependency image. */
+  lazy val dependencyImageName: Def.Initialize[String] = Def.setting {
+    fullImageName.value + "-dependencies"
+  }
+
+  //////////////////////////////////////////////////////////////////////////////////////////////////
+  // Common settings used across tasks.
+  //////////////////////////////////////////////////////////////////////////////////////////////////
+
+  /** The src/main directory. This does not appear to be a setting key in sbt. */
+  lazy val sourceMain: Def.Initialize[File] = Def.setting {
+    new File(Keys.sourceDirectory.value, "main")
+  }
+
+  /** The location of the docker target directory containing all output for the docker plugin. */
+  lazy val dockerTargetDir: Def.Initialize[File] = Def.setting {
+    new File(Keys.target.value, "docker")
+  }
+
+  /** The location of the staged dependency image. */
+  lazy val dependencyImageDir: Def.Initialize[File] = Def.setting {
+    new File(dockerTargetDir.value, "dependencies")
+  }
+
+  /** The location of the staged main image. */
+  lazy val mainImageDir: Def.Initialize[File] = Def.setting {
+    new File(dockerTargetDir.value, "main")
+  }
+
+  /** The location of the staged dependency image's library directory, containing all staged jars.
+    */
+  lazy val dependencyLibDir: Def.Initialize[File] = Def.setting {
+    new File(dependencyImageDir.value, "lib")
+  }
+
+  /** The location of the staged dependency image's Dockerfile. */
+  lazy val dependencyDockerfile: Def.Initialize[File] = Def.setting {
+    new File(dependencyImageDir.value, "Dockerfile")
+  }
+
+  //////////////////////////////////////////////////////////////////////////////////////////////////
+  // Definitions for plugin tasks.
+  //////////////////////////////////////////////////////////////////////////////////////////////////
+
+  /** Task to stage a docker image containing the dependencies of the current project. This is used
+    * to build a base image for the main project image.
+    *
+    * The result of this task is the directory containing the staged image. The directory will
+    * contain a Dockerfile to build the image and a `lib` folder with all dependency jars.
+    */
+  lazy val dependencyStageDef: Def.Initialize[Task[File]] = Def.task {
+    val logger = Keys.streams.value.log
+    logger.info("Staging dependency image ...")
+
+    // Create the destination directory.
+    val imageDirectory = dependencyImageDir.value
+    IO.createDirectory(imageDirectory)
+    val lib = dependencyLibDir.value
+    IO.createDirectory(lib)
+
+    // Create the Dockerfile for the dependency image.
+    val dockerfileContents = s"""
+      |FROM ${dockerImageBase.value}
+      |
```

Comment:
Nit-pick: Recommend removing the empty-line padding, since this file is so short.